### PR TITLE
Change log for WPCS version 1.2.0

### PR DESCRIPTION
**Pull request by jrfnl**
* Release date set at this Monday **November 12th**.
* Includes all currently merged changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,55 @@ This projects adheres to [Semantic Versioning](https://semver.org/) and [Keep a
...
+    - This sniff is disabled by default and intended as a utility tool.
+    - The sniff will fix the text domains in all I18n function calls as well as in a plugin/theme `Text Domain:` header.
+    - Passing the following properties will activate the sniff:
+        - `old_text_domain`: an array with one or more (old) text domain names which need to be replaced;
+        - `new_text_domain`: the correct (new) text domain as a string.
+- The `WordPress.NamingConventions.PrefixAllGlobals` sniff will now also verify that namespace names use a valid prefix.
+    * The sniff allows for underscores and (other) non-word characters in a passed prefix to be converted to namespace separators when used in a namespace name.
+        In other words, if a prefix of `my_plugin` is passed as a value to the `prefixes` property, a namespace name of both `My\Plugin` as well as `My_Plugin\\`, will be accepted automatically
+    * Passing a prefix property value containing namespace separators will now also be allowed and will no longer trigger a warning.
+- `WordPress` to the prefix blacklist for the `WordPress.NamingConventions.PrefixAllGlobals` sniff.
```
Comment (ntwb) [inline CHANGELOG.md]: Nit: Preclude the exception with `Add`, i.e.:
```suggestion
- Add `WordPress` to the prefix blacklist for the `WordPress.NamingConventions.PrefixAllGlobals` sniff.
```

Comment (jrfnl) [reply CHANGELOG.md]: The  current way of writing this is in line with the way it has been in the previous changelogs, this bullet is in the `Added` section, so the `Add(ed)` at the start of the change item is left out.

Comment (ntwb) [reply CHANGELOG.md]: Cool 😎

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,55 @@ This projects adheres to [Semantic Versioning](https://semver.org/) and [Keep a
...
+## [1.2.0] - 2018-11-12
+
+### Added
+- New `WordPress.PHP.TypeCasts` sniff to the `WordPress-Core` ruleset.
+    This new sniff checks that PHP type casts are:
+    * lowercase;
+    * short form, i.e. `(bool)` not `(boolean)`;
+    * normalized, i.e. `(float)` not `(real)`;
+   Additionally, the new sniff discourages the use of the `(unset)` and `(binary)` type casts.
+- New `WordPress.Utils.I18nTextDomainFixer` sniff which can compehensively replace/add `text-domain`s in a plugin or theme.
```
Comment (GaryJones) [inline CHANGELOG.md]: Change: 
`text-domain`s
to:
text domains

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,55 @@ This projects adheres to [Semantic Versioning](https://semver.org/) and [Keep a
...
+    Important notes:
+    - This sniff is disabled by default and intended as a utility tool.
+    - The sniff will fix the text domains in all I18n function calls as well as in a plugin/theme `Text Domain:` header.
+    - Passing the following properties will activate the sniff:
+        - `old_text_domain`: an array with one or more (old) text domain names which need to be replaced;
```
Comment (GaryJones) [inline CHANGELOG.md]: Change:
domain names
to:
domains

File: CHANGELOG.md (hunk 2 above)
Comment (jrfnl) [reply CHANGELOG.md]: In this case, it's within `code` tags, which is why I let it mirror what you'd expect to find in code.

**Comment by jrfnl** [comment]
Updated the commit with the fixes (except for the one requested fix where I left a reply).

If we're all happy with it, we could release earlier ?

**Comment by GaryJones** [comment]
> If we're all happy with it, we could release earlier ?

Fine with me, although I may try and look at #1526 to see if that can be done before this release.

**Comment by jrfnl** [comment]
Ok, we'll release on the 12th or when a fix for #1526 has been pulled and merged, whichever comes first.

**Comment by jrfnl** [comment]
Rebased and added an entry to cover #1530.

If there are no objections and at least one approval for #1528, I'll release WPCS 1.2.0 early tomorrow.

**Pull request merged**